src/QsCompiler/Core/SyntaxGenerator.fs

```diff
@@ -283,10 +299,12 @@ module SyntaxGenerator =
         | _ -> ArgumentException "expecting the given argument tuple to be a QsTuple" |> raise
         |> QsTuple
 
+    /// <summary>
     /// Given a typed expression that is used as argument to an operation and a typed expression for the control qubits,
     /// combines them to a suitable argument for the controlled version of the originally called operation under the assumption that the argument was correct.
     /// The range information for the built expression is set to Null.
-    /// Throws an ArgumentException if the given expression for the control qubits is a valid type but not of type Qubit[].
+    /// </summary>
+    /// <exception cref="ArgumentException"><paramref name="ctlQs"/> is a valid type but not of type <see cref="T:Qubit[]"/>.</exception>
```

Comment:
I think `Qubit[]` here is actually referring to the Q# type, so `ArrayType Qubit` in F# (but I'm not sure if it's worth trying to create a `<see>` tag for that).

Reply:
Just pushed a separate commit which adds `<see>` tags for both `ArrayType` and `Qubit`.

I don't believe there's a better way to compose them in this case, unless we were to refer directly to `QubitArray`, but this is likely not possible (or at least not hip) since it's a private member.

If you feel strongly that this is not readable, we can easily drop the commit before merging.

Reply:
I think the current version looks fine, thanks!